### ISSUE TITLE
Only those in LAN may access Web UI ipv6 support

### ISSUE THREAD

**Issue by axet** [LizardByte/Sunshine#1775]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

Config option "Only those in LAN may access Web UI" working incorrectly. Current implementation only only check IPv4 for unroutable address. But here is also IPv6 with unrotable (local link) addresses. You can see all unroutable addresses in wikipeida:

* https://ru.wikipedia.org/wiki/IPv4
* https://ru.wikipedia.org/wiki/IPv6

### Host Operating System

Linux

### Operating System Version

bookworm

### Architecture

64 bit

### Sunshine commit or version

0.21.0

### Package

Linux - flatpak

### GPU Type

AMD

### GPU Model

rx 6600

### GPU Driver/Mesa Version

0000

### Config

```Shell
000
```

### Relevant log output

```shell
000
```

**Comment by cgutman** [comment]
Yes, this is a known limitation of our IPv6 support. We do not currently have any method of enumerating interfaces and addresses in our platform-specific backends, so we can't tell if a given address is on-link or not.

**Comment by axet** [comment]
No need for that. Listening 0.0.0.0 - makes you ipv4 only. Listening :: makes you ipv4/ipv6. You have peer address which can be validated for local node connection / remote connection.

**Comment by cgutman** [comment]
I don't understand what you're suggesting. Yes, we can listen on 0.0.0.0 or :: and that's exactly what we already do depending on whether you selected IPv4-only or IPv6+IPv4 in Sunshine's configuration.

As I mentioned, we need to know whether the peer address is on-link or not. Unlike IPv4, you can't easily know that based on the address alone. You need to enumerate the assigned addresses and subnet masks on local interfaces to determine on-link destinations, and that isn't implemented yet.

**Comment by axet** [comment]
Not true. Unless you also want to add routable local IPv6 addresses, which BTW I do not recommend. IPv6 not a bit different from IPv4.

* https://ru.wikipedia.org/wiki/IPv4
* https://ru.wikipedia.org/wiki/IPv6

Check unroutable addesses (local link)

**Comment by Diatrus** [comment]
I've been having trouble with remote ipv6 as well, and I was surprised to see sunshine only opens on my link-local address? Looks like this has been an issue for a while. I'm surprised this hasn't been prioritized more, as sunshine should be chasing a global address to advertise on, rather than a link-local.
